chore(release): prepare for 1.12.0 (#1034)
chatmail/relay: CHANGELOG.md

# Changelog for chatmail deployment

## [1.12.0] - 2026-07-31

### Breaking Changes

> - [**breaking**] Introduce configurable system limits to reject new address creation and limit imap/smtp connections.

Reviewer: Lets be explicit here and mention that we lowered the limits from 50k to 10k for Dovecot and 5k to 1k for Postfix to provide a lower bound for small relays, to draw attention operators of larger relays to the need to adjust their configs proactively.

### Features

- Reduce maximal_queue_lifetime from 5d to 2d
- Disable negative cache in unbound (#992)
- *(mtail)* Add incoming_mailer_daemon_mail_count
- *(postfix)* Disable processing of MIME headers
- *(dovecot)* Advertise privacy_mail as admin contact, drop server comment

### Bug Fixes

- Set relay restrictions per smtpd service with default reject
- Reduce maxproc for filtermail-transport LMTP client to 500
- Core 2.50.0 does not have delete_server_after config anymore.
- Check if all required ports are available for filtermail (#983)
- Always deploy unbound.conf.d/chatmail.conf (#993)
- Expire empty directories (#994)
- Crypt-r dependency was declared for wrong Python version

> - Still overwrite /etc/resolv.conf if it is a symbolic link

Reviewer: ```suggestion
- Always overwrite /etc/resolv.conf, even if it is a symbolic link
```

- Pass kwargs to files.put()
- List Iroh proxy endpoints used by 0.35 and 1.0, drop stale /relay/probe from earlier versions
- Fix port discovery when ss -tulpn shows dovecot before stats

### Documentation

- Add scripts/initenv.sh to upgrade instructions
- Update overview diagrams (#995)
- *(overview)* Remove mermaid styles from 'Accepting and delivering mail' (#1009)
- *(README.md)* Clarify security enforcement (#1011)

### Miscellaneous Tasks

> - Auto-trigger docker build on release tag push

Reviewer: ```suggestion
- (ci) Auto-trigger docker build on release tag push
```

- *(acmetool)* Update let's encrypt ToS link
- Update Let's Encrypt Subscriber Agreement to 1.8
- *(ci)* Update doc staging upload path

> - Fix docs upload path

Reviewer: ```suggestion
- (ci) Fix docs upload path
```
